server/chat-plugins/poll.ts

```diff
@@ -350,27 +350,33 @@ export const commands: ChatCommands = {
 
 		clearqueue: 'deletequeue',
 		deletequeue(target, room, user, connection, cmd) {
-			room = this.requireRoom();
-			this.checkCan('mute', null, room);
-			if (!room.minorActivityQueue) {
+			target = target.trim();
+			const [slotString, roomid] = target.split(',');
+			const Room = roomid ? Rooms.search(roomid) : room;
```

Comment:
This is additionally not the fix you want. This does not work.
This is what you want instead.
```ts
const [slotString, roomid, update] = target.split(',');
room = roomid ? Rooms.search(roomid) : room;
room = this.requireRoom();

```

None of these other changes are necessary.

Reply:
If I add only this code, there is an error: http://prntscr.com/w99uye

Reply:
I mean that's pretty simple to fix.
Just make it into:
room = roomid ? Rooms.search(roomid) : room;
this.room = room;
room = this.requireRoom();

Reply:
![](https://image.prntscr.com/image/BlQUDG9DSwSgIeI5LTjDDg.png)

Sorry I'm trying to learn at the same time

Reply:
```suggestion
					clearqueue: 'deletequeue',
		deletequeue(target, room, user, connection, cmd) {
			const [slotString, roomid, update] = target.split(',').map(i => i.trim());
			const targetRoom = roomid ? Rooms.search(roomid) : room;
			if (!targetRoom) return this.errorReply(`Invalid room.`);
			if (!targetRoom.persist) {
				return this.errorReply("This command is unavailable in temporary rooms.");
			}
			this.room = targetRoom;
			room = this.requireRoom();
			this.checkCan('mute', null, room);
			if (!targetRoom.minorActivityQueue) {
				return this.errorReply(this.tr`The queue is already empty.`);
			}
			if (cmd === 'deletequeue' && targetRoom.minorActivityQueue.length !== 1 && !target) {
				return this.parse('/help deletequeue');
			}
			if (!target) {
				targetRoom.minorActivityQueue = null;
				this.modlog('CLEARQUEUE');
				this.sendReply(this.tr`Cleared poll queue.`);
			} else {
				const [slotString, roomid, update] = target.split(',');
				const slot = parseInt(slotString);
				const curRoom = roomid ? (Rooms.search(roomid) as ChatRoom | GameRoom) : room;
				if (!curRoom) return this.errorReply(this.tr`Room "${roomid}" not found.`);
				if (isNaN(slot)) {
					return this.errorReply(this.tr`Can't delete poll at slot ${slotString} - "${slotString}" is not a number.`);
				}
				if (!targetRoom.minorActivityQueue[slot - 1]) return this.errorReply(this.tr`There is no poll in queue at slot ${slot}.`);

				curRoom.minorActivityQueue!.splice(slot - 1, 1);
				if (!curRoom.minorActivityQueue?.length) curRoom.minorActivityQueue = null;

				curRoom.modlog({
					action: 'DELETEQUEUE',
					loggedBy: user.id,
					note: slot.toString(),
				});
				curRoom.sendMods(this.tr`(${user.name} deleted the queued poll in slot ${slot}.)`);
				curRoom.update();
				if (update) this.parse(`/j view-pollqueue-${curRoom}`);
			}
		},
```
It's good like that ?

Reply:
Remove line 373. Duplicated code.

Reply:
You also need the "update" variable to update the page if it's there.

Reply:
At line 391 you check for if (roomid) (parse join), but that should actually be if (update), as well.